plugins/modules/iosxr_ntp_global.py

```diff
@@ -0,0 +1,1077 @@
+          suboptions:
+            aging_time:
+              type: int
+              description: Aging time in hours.
+            file:
+              description: File for drift values.
+              type: str
+        interfaces:
+          type: list
+          elements: dict
+          description: Configure NTP on an interface.
+          suboptions:
+            name:
+              type: str
+              description: Name of the interface.
+            vrf:
+              type: str
+              description: Name of the vrf.
+            broadcast_client:
+              type: bool
+              description: Listen to NTP broadcasts
+            broadcast_destination:
+              type: str
+              description: Configure broadcast destination address.
+            broadcast_key:
+              type: int
+              description: Broadcast key number.
+            broadcast_version:
+              type: int
+              description: <2-4>  NTP version number.
+            multicast_key:
+              type: int
+              description: Configure multicast authentication key.
+            multicast_ttl:
+              type: int
+              description: Configure TTL to use.
+            multicast_client:
+              type: str
+              description: Configure multicast client
+            multicast_destination:
+              type: str
+              description: Configure multicast destination
+            multicast_version:
+              type: int
+              description: <2-4>  NTP version number.
+        ipv4: &ip
+          description: Mark the dscp/precedence bit for ipv4 packets.
+          type: dict
+          suboptions:
+            dscp:
+              description: Set IP DSCP (DiffServ CodePoint).Please refer vendor document for valid entries.
+              type: str
+            precedence:
+              description: Set precedence Please refer vendor document for valid entries.
+              type: str
+              choices: [ "critical", "flash", "flash-override", "immediate", "internet", "network", "priority", "routine" ]
+        ipv6: *ip
+        log_internal_sync:
+          type: bool
+          description: Logs internal synchronization changes.
+        master:
+          type: int
+          description: Use NTP as clock source with stratum number <1-15>
+        max_associations:
+          type: int
+          description: <0-4294967295>  Number of associations.
+        passive:
+          type: bool
+          description: Enable the passive associations.
+        trusted_keys:
+          type: list
+          elements: dict
+          description: list of Key numbers for trusted time sources.
+          suboptions:
+            key_id:
+              type: int
+              description: Key numbers for trusted time sources.
+        update_calendar:
+          type: bool
+          description: Periodically update calendar with NTP time.
+        source:
```

Comment:
If this is the source interface, can we name it as `source_interface`?

Reply:
ack